crates/mem-pool/src/pool.rs

```diff
@@ -783,34 +783,39 @@ impl MemPool {
 
         // check expire
         let mut force_expired = false;
-        let mut tasks = Vec::with_capacity(self.pending_deposits.len());
         for deposit in &self.pending_deposits {
             // check is handled by current block
             if processed_deposit_requests.contains(&deposit.request) {
                 force_expired = true;
                 break;
             }
-
-            // query deposit live cell
-            tasks.push(self.provider.get_cell(deposit.cell.out_point.clone()));
         }
 
-        // check cell is available
-        for task in tasks {
-            match task.await? {
-                Some(cell_with_status) => {
-                    if cell_with_status.status != CellStatus::Live {
+        // check whether the deposited cells are live
+        if !force_expired {
+            for deposit in &self.pending_deposits {
+                match self
+                    .provider
+                    .get_cell(deposit.cell.out_point.clone())
+                    .await?
+                {
+                    Some(CellWithStatus {
+                        status: CellStatus::Live,
+                        ..
+                    }) => {}
+                    _ => {
                         force_expired = true;
                         break;
                     }
                 }
-                None => {
-                    force_expired = true;
-                    break;
-                }
             }
         }
 
+        if force_expired {
+            log::debug!("[mem-pool] forced clear pending deposits");
+            self.pending_deposits.clear();
+        }
```

Comment:
Note for reviewers, when `force_expired` is `true`, we should clear `self.pending_deposits` before making the `safe_expired` prediction. Otherwise `safe_expired` will always be `false` when `force_expired` is equal to `true`.